devdoc/constbuffer_requirements.md

```diff
@@ -36,11 +36,11 @@ MOCKABLE_FUNCTION(, CONSTBUFFER_HANDLE, CONSTBUFFER_CreateWithMoveMemory, unsign
 
 MOCKABLE_FUNCTION(, CONSTBUFFER_HANDLE, CONSTBUFFER_CreateWithCustomFree, const unsigned char*, source, size_t, size, CONSTBUFFER_CUSTOM_FREE_FUNC, customFreeFunc, void*, customFreeFuncContext);
 
-MOCKABLE_FUNCTION(, CONSTBUFFER_HANDLE, CONSTBUFFER_Clone, CONSTBUFFER_HANDLE, constbufferHandle);
+MOCKABLE_FUNCTION(, void, CONSTBUFFER_IncRef, CONSTBUFFER_HANDLE, constbufferHandle);
 
-MOCKABLE_FUNCTION(, const CONSTBUFFER*, CONSTBUFFER_GetContent, CONSTBUFFER_HANDLE, constbufferHandle);
+MOCKABLE_FUNCTION(, void, CONSTBUFFER_DecRef, CONSTBUFFER_HANDLE, constbufferHandle);
 
-MOCKABLE_FUNCTION(, void, CONSTBUFFER_Destroy, CONSTBUFFER_HANDLE, constbufferHandle);
```

Comment:
>CONSTBUFFER_Destroy [](start = 26, length = 19)

Removing destroy is probably also not the most desirable.  I don't have a horse in the race on this one, but it is causing cascading changes through the repo.

Reply:
I sense fear of change. But fear of change should always be overcome with the desire to progress :-).

---
In reply to: [251144218](https://github.com/Azure/azure-c-shared-utility/pull/262#discussion_r251144218) [](ancestors = 251144218)

Reply:
Technically Destroy is DecRef. Any ref counted object can only have DecRef, because noone can explictly free the memory for the object. A user can only "decrement ref count and if 0 then please free memory". Thus DecRef is the appropriate thing to have for an object that has IncRef.

And this is really a simple rename that should not panic anyone.

---
In reply to: [251150747](https://github.com/Azure/azure-c-shared-utility/pull/262#discussion_r251150747) [](ancestors = 251150747,251144218)